src/coreclr/gcinfo/gcinfoencoder.cpp

```diff
@@ -776,12 +774,14 @@ void GcInfoEncoder::SetReversePInvokeFrameSlot(INT32 spOffset)
     m_ReversePInvokeFrameSlot = spOffset;
 }
 
+#ifndef TARGET_X86
 void GcInfoEncoder::SetReturnKind(ReturnKind returnKind)
```

Comment:
This is only used on Linux x86. Is there something fundamental that prevents Linux x86 to be switched to the same plan as the other GC encoder platforms? It would make the change simpler.

I know that there is no testing for Linux x86. I would not be worried about it. Since there is no testing, Linux x86 is likely broken anyway.

Reply:
Hmm. I actually wanted to make this "ifdef", not "ifndef". How does this even pass tests?

x86 GC info is not supposed to be changed.

Reply:
Ah. This is still the non-x86 GC encoder. The confusion was because this is a part where non-x86 encoder had an ifdef for TARGET_X86, so I assumed it is a shared path between two encoders. 
It is not. The reason for original ifdef was just because X86 would have a different default state. 

Regardless x86 or not, `m_ReturnKind` is not going to be used by the non-legacy GC encoder. 
This does not need to be ifdef'd. It can be removed. There is indeed nothing special about X86 here.

Reply:
I think my lack of knowledge about GC encoder history shows here.